### PR TITLE
Add CloudBuild steps to run image tests against the signatures discovery feature

### PR DESCRIPTION
**Pull request by yawangwang**
Added CloudBuild steps for both hardened image and debug image to run signature discovery tests:
1. use cosign to sign the container,
2. launch a VM with signed container image support,
3. check VM serial logs for any found signatures.

### CODE REVIEW

File: launcher/image/test/test_discover_signatures.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  - |
+    # Unpadded base64 encoding on the CloudKMS public key
+    pub=$(cosign public-key --key gcpkms://projects/confidential-space-images-dev/locations/global/keyRings/cosign-test/cryptoKeys/ecdsa/cryptoKeyVersions/1 | openssl base64)
+    pub=$(echo $pub | tr -d '[:space:]' | sed 's/[=]*$//')
+    # Use cosign sign
+    export COSIGN_REPOSITORY=${_SIGNATURE_REPO}
+    cosign sign --key gcpkms://projects/confidential-space-images-dev/locations/global/keyRings/cosign-test/cryptoKeys/ecdsa/cryptoKeyVersions/1 ${_WORKLOAD_IMAGE} -a dev.cosignproject.cosign/sigalg=ECDSA_P256_SHA256 -a dev.cosignproject.cosign/pub=$pub
+- name: 'gcr.io/cloud-builders/gcloud'
+  id: CreateVM
+  waitFor: ['SignContainer']
```
Comment (JoshuaKrstic) [inline launcher/image/test/test_discover_signatures.yaml:27]: If no waitFor is specified the step will wait for ALL previous steps to complete.
Consider removing this

Comment (JoshuaKrstic) [reply launcher/image/test/test_discover_signatures.yaml:27]: Applies to other steps as well

File: launcher/image/test/test_discover_signatures.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  id: CheckFailure
+  waitFor: ['CleanUp']
+  entrypoint: 'bash'
+  env:
+  - 'BUILD_ID=$BUILD_ID'
+  args: ['check_failure.sh']
+
+options:
+  pool:
+    name: 'projects/confidential-space-images-dev/locations/us-west1/workerPools/cs-image-build-vpc'
```
Comment (JoshuaKrstic) [inline launcher/image/test/test_discover_signatures.yaml:74]: nit: Please add a new line at the EOF

**Review by JoshuaKrstic — APPROVED** [review]

File: launcher/image/test/test_discover_signatures.yaml (hunk 1 above)
Comment (yawangwang) [reply launcher/image/test/test_discover_signatures.yaml]: Done.

File: launcher/image/test/test_discover_signatures.yaml (hunk 2 above)
Comment (yawangwang) [reply launcher/image/test/test_discover_signatures.yaml]: Done.

**Pull request merged**